From: Alexis Murzeau <amubtdx@gmail.com>
Date: Sat, 4 Apr 2020 15:16:10 +0200
Subject: docs: replace deprecated source_parsers

With app.add_source_parser in a setup() function.
This function is taken from the recommonmark 0.6 extension.

This patch cannot be upstream-ed as-is as the solution would instead to
use recommonmark as a normal extension.
I use a script here because Debian has recommonmark 0.4 and that version
doesn't have the required code to be used as an extension.

Steps to fix this:
 - Upgrade recommonmark package in Debian
   (https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=955180)
 - Update the patch to use recommonmark as an extension instead
 - Upstream that patch

Forwarded: todo
Author: Alexis Murzeau <amubtdx@gmail.com>
---
 docs/conf.py | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/docs/conf.py b/docs/conf.py
index 7b05bb3..68cfa98 100644
--- a/docs/conf.py
+++ b/docs/conf.py
@@ -6,6 +6,17 @@ import sys
 
 from streamlink import __version__ as streamlink_version
 
+# The config variable "source_parsers" is deprecated. Use setup() function to implement this instead.
+def setup(app):
+    import sphinx
+    from recommonmark.parser import CommonMarkParser
+
+    if sphinx.version_info >= (1, 8):
+        app.add_source_suffix('.md', 'markdown')
+        app.add_source_parser(CommonMarkParser)
+    elif sphinx.version_info >= (1, 4):
+        app.add_source_parser('.md', CommonMarkParser)
+
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
@@ -23,12 +34,6 @@ extensions = ['sphinx.ext.autodoc', 'ext_argparse', 'ext_github', 'ext_releasere
 # Add any paths that contain templates here, relative to this directory.
 templates_path = ['_templates']
 
-# The suffix of source filenames.
-source_suffix = ['.rst', '.md']
-source_parsers = {
-   '.md': 'recommonmark.parser.CommonMarkParser',
-}
-
 # The encoding of source files.
 #source_encoding = 'utf-8-sig'
 
